src/ui/utils/sources.ts

```diff
@@ -1,118 +1,63 @@
   const newSourcesById: Record<string, newSource> = {};
-  const byKind = {} as Record<SourceKind, newSource[]>;
-
-  // Canonical links can go across multiple links
-  const findCanonicalId = (id: string) => {
-    let current = id;
-    let nextNode = canonical.from(current)?.[0];
-
-    while (nextNode && nextNode !== current) {
-      current = nextNode;
-      nextNode = canonical.from(current)?.[0];
-    }
-
-    return current;
-  };
-
-  const generated = newGraph();
-
-  // We'll iterate through the whole list once, and perform three setup operations per source:
-  // 1) Creating a normalized lookup table by source ID
-  // 2) Grouping sources by kind
-  // 3) Connected nodes based on generated source IDs
-  for (let source of newSources) {
-    // start by adding this source to the lookup table
-    newSourcesById[source.sourceId] = source;
-
-    // Also group by kind
-    if (!byKind[source.kind]) {
-      byKind[source.kind] = [];
-    }
-
-    byKind[source.kind].push(source);
-
-    // We handle pretty-printed (pp) files and their generated links a little
-    // differently. Because Replay makes the pp sources, their structure is
-    // predictable. All pp sources will have one generatedSourceId, and it will
-    // be the minified source.
-    if (source.kind === "prettyPrinted") {
-      continue;
-    }
-
-    for (let generatedId of source.generatedSourceIds || []) {
-      // source.generatedSourceIds?.map(generatedId => {
-      generated.connectNode(source.sourceId, generatedId);
-    }
-  }
-
-  // Sources are processed by kind
-
-  const inlineScripts = byKind["inlineScript"] || [];
-  for (let source of inlineScripts) {
-    canonical.connectNode(source.sourceId, generated.to(source.sourceId)![0]);
-  }
-
-  const sourceMapped = byKind["sourceMapped"] || [];
-  for (let source of sourceMapped) {
-    canonical.connectNode(generated.from(source.sourceId)![0], source.sourceId);
-  }
+  const corresponding = new ArrayMap<string, string>();
+  const original = new ArrayMap<string, string>();
+  const prettyPrinted: Record<string, string> = {};
 
   // TODO @jcmorrow: remove this once we include the contentHash with prettyPrinted sources
   const contentHashForSource = (source: newSource) => {
     return source.kind === "prettyPrinted"
       ? newSourcesById[source.generatedSourceIds![0]].contentHash
       : source.contentHash;
   };
+  const isSourceMapped = (source: newSource) => {
+    return source.kind === "prettyPrinted"
+      ? newSourcesById[source.generatedSourceIds![0]].kind === "sourceMapped"
+      : source.kind === "sourceMapped";
+  };
 
   const keyForSource = (source: newSource): string => {
     return `${source.kind}:${source.url!}:${contentHashForSource(source)}`;
   };
 
   for (let source of newSources) {
-    const key = keyForSource(source);
-    if (corresponding[key] === undefined) {
-      corresponding[key] = [];
+    newSourcesById[source.sourceId] = source;
+  }
```

Comment:
If we don't need to do the corresponding check I think it would be easier to read this as
`const newSourcesById = groupBy(newSources, "sourceId)` but I don't feel strongly about it.

Reply:
Failing making that change we should at least have line 30 be `for (const ...` rather than `for (let ...`

Reply:
`groupBy` would create a `Dictionary<newSource[]>`, but we know that there is only one source per sourceId, so I prefer not using that (though I don't feel strongly about it either).
I'll replace all `for (let ...` with `for (const ...`.